File: iosApp/iosApp/ui/components/MediaCarousel.swift
import SwiftUI
import Shared

/**
 * Horizontal carousel for displaying multiple media items
 * Shows dot indicators when there are multiple items
 */
struct MediaCarousel: View {
    let media: [PostMedia]
    let onMediaClick: () -> Void
    let isVisible: Bool
    
    @State private var currentPage = 0
    
    var body: some View {
        if media.isEmpty {
            EmptyView()
        } else if media.count == 1 {
            // Single media item - no carousel needed
            MediaItemView(
                media: media[0],
                isVisible: isVisible,
                onTap: onMediaClick
            )
        } else {
            // Multiple media items - show carousel with TabView
            ZStack(alignment: .bottom) {
                TabView(selection: $currentPage) {
                    ForEach(Array(media.enumerated()), id: \.offset) { index, item in
                        MediaItemView(
                            media: item,
                            isVisible: isVisible && (index == currentPage),
                            onTap: onMediaClick
                        )
                        .tag(index)
                    }
                }
                .tabViewStyle(.page(indexDisplayMode: .never))
                .frame(height: 220)
                
                // Dot indicators overlaid on bottom of media - transparent background
                HStack(spacing: 6) {
                    ForEach(0..<media.count, id: \.self) { index in
                        Circle()
                            .fill(index == currentPage ? Color.white : Color.white.opacity(0.5))
                            .frame(width: index == currentPage ? 6 : 5, height: index == currentPage ? 6 : 5)
                    }
                }
                .padding(.horizontal, 12)
                .padding(.vertical, 8)
            }
        }
    }
}

File: composeApp/src/androidMain/kotlin/com/kissangram/ui/home/components/MediaCarousel.kt
package com.kissangram.ui.home.components

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.pager.HorizontalPager
import androidx.compose.foundation.pager.rememberPagerState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Surface
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.kissangram.model.PostMedia

/**
 * Horizontal carousel for displaying multiple media items
 * Shows dot indicators when there are multiple items
 * Uses HorizontalPager to match iOS TabView behavior (one item at a time)
 */
@Composable
fun MediaCarousel(
    media: List<PostMedia>,
    modifier: Modifier = Modifier,
    onMediaClick: () -> Unit,
    isVisible: Boolean = true
) {
    if (media.isEmpty()) {
        return
    }
    
    // Single media item - no carousel needed
    if (media.size == 1) {
        MediaItemView(
            media = media.first(),
            modifier = modifier,
            isVisible = isVisible,
            onTap = onMediaClick
        )
        return
    }
    
    // Multiple media items - show carousel with HorizontalPager
    val pagerState = rememberPagerState(pageCount = { media.size })
    
    Box(modifier = modifier) {
        HorizontalPager(
            state = pagerState,
            modifier = Modifier
                .fillMaxWidth()
                .height(220.dp) // Match iOS height
        ) { page ->
            MediaItemView(
                media = media[page],
                modifier = Modifier.fillMaxSize(),
                isVisible = isVisible && (page == pagerState.currentPage),
                onTap = onMediaClick
            )
        }
        
        // Dot indicators overlaid on bottom of media - transparent background
        Row(
            modifier = Modifier
                .align(Alignment.BottomCenter)
                .fillMaxWidth()
                .padding(horizontal = 12.dp, vertical = 8.dp),
            horizontalArrangement = Arrangement.Center
        ) {
            repeat(media.size) { index ->
                val isSelected = index == pagerState.currentPage
                Surface(
                    modifier = Modifier
                        .size(if (isSelected) 6.dp else 5.dp)
                        .padding(horizontal = 3.dp),
                    shape = CircleShape,
                    color = if (isSelected) {
                        Color.White
                    } else {
                        Color.White.copy(alpha = 0.5f)
                    }
                ) {}
            }
        }
    }
}
